File: app/src/main/java/com/papco/sundar/papcortgs/screens/password/PasswordDialog.java
package com.papco.sundar.papcortgs.screens.password;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.appcompat.app.AlertDialog;
import android.text.InputType;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.Toast;

import com.papco.sundar.papcortgs.R;

public class PasswordDialog extends DialogFragment {

    public static final int CODE_RECEIVERS=1;
    public static final int CODE_SENDERS=2;
    private final String KEY_SAVE_CODE="REQUEST_CODE";

    private PasswordCallback callback;
    private final String PASSWORD="papco1954";
    private int code=1;
    private EditText password;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(savedInstanceState!=null){

            code=savedInstanceState.getInt(KEY_SAVE_CODE);
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        View view=getActivity().getLayoutInflater().inflate(R.layout.new_group,null);
        password=view.findViewById(R.id.new_tag_editText);
        password.setInputType(InputType.TYPE_CLASS_TEXT |
                InputType.TYPE_TEXT_VARIATION_PASSWORD);
        AlertDialog.Builder builder=new AlertDialog.Builder(getActivity());
        builder.setTitle("Enter password");
        builder.setView(view);
        builder.setNegativeButton("CANCEL",null);
        builder.setPositiveButton("OK", null);
        final AlertDialog dialog=builder.create();
        dialog.setOnShowListener(new DialogInterface.OnShowListener() {
            @Override
            public void onShow(DialogInterface dialogInterface) {
                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        validatePassword(dialog,password.getText().toString());
                    }
                });
            }
        });

        showKeyboard();

        return dialog;


    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt(KEY_SAVE_CODE,code);
    }


    private void showKeyboard(){

        InputMethodManager imgr=(InputMethodManager)getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
        if(password!=null) {
            password.requestFocus();
            //imgr.showSoftInput(password, InputMethodManager.SHOW_IMPLICIT);
            imgr.toggleSoftInput(InputMethodManager.SHOW_FORCED, InputMethodManager.HIDE_IMPLICIT_ONLY);
        }

    }

    public void setRequestCode(int code){
        this.code=code;
    }

    private void validatePassword(AlertDialog dialog, String enteredPassword) {

        if(enteredPassword.equals(PASSWORD)){
            dialog.dismiss();
            try{

                callback=(PasswordCallback)getActivity();
                callback.onPasswordOk(code);

            }catch (Exception e){

                Toast.makeText(getActivity(),"Did you forget to implement PasswordCallback in calling activity?",Toast.LENGTH_SHORT).show();

            }
        }else{
            Toast.makeText(getActivity(),"Invalid password. Try again",Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/papco/sundar/papcortgs/screens/password/PasswordDialog.kt
package com.papco.sundar.papcortgs.screens.password

import android.content.Context
import android.os.Bundle
import android.text.InputType
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.Toast
import androidx.navigation.fragment.findNavController
import com.papco.sundar.papcortgs.R
import com.papco.sundar.papcortgs.common.ResultDialogFragment
import com.papco.sundar.papcortgs.databinding.DialogTextInputBinding

class PasswordDialog : ResultDialogFragment() {

    companion object {

        private const val KEY_CODE = "REQUEST_CODE"
        private const val PASSWORD = "papco1954"
        const val TAG = "PasswordDialog"

        fun getInstance(code: Int): PasswordDialog {
            return PasswordDialog().apply {
                val args = Bundle()
                args.putInt(KEY_CODE, code)
                arguments = args
            }
        }

        fun getArgumentBundle(code: Int): Bundle {
            return Bundle().also {
                it.putInt(KEY_CODE, code)
            }
        }
    }

    private var _viewBinding: DialogTextInputBinding? = null
    private val viewBinding: DialogTextInputBinding
        get() = _viewBinding!!


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        _viewBinding = DialogTextInputBinding.inflate(inflater, container, false)
        return viewBinding.root
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViews()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _viewBinding = null
    }


    private fun initViews() {
        viewBinding.textInputDialogTitle.text = "Enter Password"
        viewBinding.textInputDialogText.inputType =
            InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
        viewBinding.textInputDialogText.hint="Password"

        viewBinding.textInputDialogOk.setOnClickListener {
            if (enteredPassword() == PASSWORD) {
                if (dispatchResult(true, getCode())) dismiss()
            } else Toast.makeText(
                requireContext(),
                getString(R.string.incorrect_password),
                Toast.LENGTH_SHORT
            ).show()
        }

        viewBinding.textInputDialogCancel.setOnClickListener {
            dismiss()
        }

        showKeyboard()

    }

    private fun showKeyboard() {
        val inputMethodManager =
            requireActivity().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        viewBinding.textInputDialogText.requestFocus()
        inputMethodManager.showSoftInput(viewBinding.textInputDialogText, 0)

    }


    private fun enteredPassword(): String {
        return viewBinding.textInputDialogText.text.toString().trim()
    }

    private fun getCode(): Int {
        return arguments?.getInt(KEY_CODE) ?: error("Invalid argument code in password dialog")
    }
}
